scripts/mdpow-solvationenergy

```diff
@@ -105,20 +105,59 @@ def run_gsolv(solvent, directory, **kwargs):
 
     gsolv = load_gsolv(solvent, directory, permissive=kwargs.pop('permissive',False),)
 
+    if not hasattr(gsolv, 'start'):
+        gsolv.start = kwargs.pop('start', 0)
+    if not hasattr(gsolv, 'stop'):
+        gsolv.stop = kwargs.pop('stop', None)
+    if not hasattr(gsolv, 'SI'):
+        kwargs.setdefault('SI', True)
+    else:
+        kwargs.setdefault('SI', gsolv.SI)
+
+    # for this version. use the method given instead of the one in the input cfg file
+    gsolv.method = kwargs.pop('method', 'MBAR')
+    estimator = kwargs.pop('estimator', 'alchemlyb')
+
     # do the analysis (only use keywords that are needed for FEP.analysis())
     akw = {'stride': kwargs.pop('stride', None), 'force': kwargs.pop('force', False)}
+    if not estimator in ('mdpow', 'alchemlyb'):
+        errmsg = "estimator = %r is not supported, must be 'mdpow' or 'alchemlyb'" % estimator
+        logger.error(errmsg)
+        raise ValueError(errmsg)
     if akw['force']:
         logger.info("[%(directory)s] Forcing (re)analysis of hydration free energy data", vars())
-        gsolv.analyze(**akw)
+        if estimator == 'mdpow':
+            if G.method != "TI":
+                errmsg = "Method %s is not implemented in MDPOW, use estimator='alchemlyb'" % G.method
+                logger.error(errmsg)
+                raise ValueError(errmsg)
+            gsolv.start = kwargs.pop('start', 0)
+            gsolv.stop = kwargs.pop('stop', None)
+            gsolv.SI = kwargs.pop('stop', False)
+            gsolv.analyze(**akw)
+
+        elif estimator == 'alchemlyb':
+            gsolv.analyze_alchemlyb(**akw)
     # make sure that we have data
     try:
         gsolv.results.DeltaA.Gibbs
         logger.info("[%(directory)s] Reading existing data from pickle file.", vars())
         gsolv.logger_DeltaA0()
     except (KeyError, AttributeError):   # KeyError because results is a AttributeDict
         logger.info("[%(directory)s] Analyzing %(solvent)s solvation free energy data for the first time", vars())
-        gsolv.analyze(**akw)
-
+        if estimator == 'mdpow':
+            if G.method != "TI":
+                errmsg = "Method %s is not implemented in MDPOW, use estimator='alchemlyb'" % G.method
+                logger.error(errmsg)
+                raise ValueError(errmsg)
+            gsolv.start = kwargs.pop('start', 0)
+            gsolv.stop = kwargs.pop('stop', None)
+            gsolv.SI = kwargs.pop('stop', False)
```

Comment:
This should all be removed and dealt with earlier. I don't understand why this code needs to be repeated here.